presto-iceberg/src/main/java/com/facebook/presto/iceberg/HiveTableOperations.java

```diff
@@ -327,7 +342,54 @@ public void commit(@Nullable TableMetadata base, TableMetadata metadata)
             }
             else {
                 PartitionStatistics tableStats = metastore.getTableStatistics(metastoreContext, database, tableName);
-                metastore.persistTable(metastoreContext, database, tableName, table, privileges, () -> tableStats, useHMSLock ? ImmutableMap.of() : hmsEnvContext(base.metadataFileLocation()));
+                try {
+                    metastore.persistTable(metastoreContext, database, tableName, table, privileges, () -> tableStats, useHMSLock ? ImmutableMap.of() : hmsEnvContext(base.metadataFileLocation()));
+                }
+                catch (ValidationException e) {
```

Comment:
```suggestion
                catch (InvalidObjectException e) {
```

A typo? As I see, the Iceberg's code catch `InvalidObjectException` here.